### PR TITLE
Ev 18 refactor search user page in profile

### PR DESCRIPTION
**Pull request by ghost**
### Jira

* [EV-NUM](https://chita.atlassian.net/browse/EV-NUM)

### Description

Refactor SearchUsers.js and assosiated components

### Checklist

- [x] Includes a link to JIRA ticket and all related PR (API dependencies etc.)
- [x] Includes a brief description of what has been changed
- [x] Test coverage for **new** code doesn't decrease the existing test coverage
- [x] Lint errors/warnings fixed
- [x] New code is documented if necessary
- [x] Dev Testing has been successfully finished
- [x] Reviewed by 2 developer on your team
- [x] Reviewed by 1 expert

### CODE REVIEW

File: src/components/SearchUsers/SearchUsers.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+﻿/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (andrii-kuzmichov) [inline src/components/SearchUsers/SearchUsers.js:1]: drop

File: src/components/SearchUsers/SearchUsers.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
```
Comment (andrii-kuzmichov) [inline src/components/SearchUsers/SearchUsers.js:2]: drop empty line

File: src/components/SearchUsers/SearchUsers.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { useHistory, useLocation } from "react-router-dom";
+import Pagination from "@material-ui/lab/Pagination";
+
+// import MainSearchInput from "../searchInput/MainSearchInput";
```
Comment (andrii-kuzmichov) [inline src/components/SearchUsers/SearchUsers.js]: drop

File: src/components/SearchUsers/SearchUsers.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import SearchInput from "../searchInput/SearchInput";
+import { getSearchUsers, changeFilter } from "../../actions/users/users-action";
+import SpinnerWrapper from "../../containers/spinner";
+import UserItemList from "../users/UserItemList";
+
```
Comment (andrii-kuzmichov) [inline src/components/SearchUsers/SearchUsers.js]: drop

File: src/components/SearchUsers/SearchUsers.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
+.search_users_container {
+  height: 100%;
```
Comment (andrii-kuzmichov) [inline src/components/SearchUsers/SearchUsers.scss]: constants

File: src/components/UserInfoCard/AttitudeToolTip/AttitudeToolTip.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Tooltip from "@material-ui/core/Tooltip";
+import Zoom from "@material-ui/core/Zoom";
+import "./AttitudeToolTip.scss";
+
+const AttitudeToolTip = ({ message, thumb }) => {
+  return (
+    <Tooltip title={message} placement="bottom" TransitionComponent={Zoom}>
+      <div className="retreat">
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/AttitudeToolTip/AttitudeToolTip.jsx]: extra

File: src/components/UserInfoCard/AttitudeToolTip/AttitudeToolTip.scss [hunk 7]
```diff
@@ -0,0 +1,10 @@
+.retreat {
+  width: 32px;
+  padding-top: 5px;
+  padding-right: 15px;
+  text-align: end;
+}
+
+.thumb_size {
+  font-size: 24px;
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/AttitudeToolTip/AttitudeToolTip.scss]: constants

File: src/components/UserInfoCard/UserInfoCard.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
+﻿import React from "react";
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.jsx]: drop

File: src/components/UserInfoCard/UserInfoCard.jsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import genders from "../../constants/GenderConstants";
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.jsx:4]: Do not use default export

File: src/components/UserInfoCard/UserInfoCard.jsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+import CustomAvatar from "../avatar/custom-avatar";
+import RatingAverage from "../rating/rating-average";
+import defineUserAge from "../helpers/defineUserAge";
+import getAttitudeToUser from "../helpers/getAttitudeToUser";
+import AttitudeToolTip from "./AttitudeToolTip/AttitudeToolTip";
+
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.jsx]: drop

File: src/components/users/UserItemList.jsx [hunk 11]
```diff
@@ -1,22 +1,30 @@
...
-  const renderUsers = arr => {
-    return arr.map(user => <UserInfoCard key={user.id} user={user} />);
-  };
-
   return (
     <>
-      {renderUsers(users)}
+      <GridLayout>
+        {users.map(user => (
+          <UserInfoCard key={user.id} user={user} />
```
Comment (andrii-kuzmichov) [inline src/components/users/UserItemList.jsx]: use destructuring

File: src/components/UserInfoCard/UserInfoCard.jsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+import "./UserInfoCard.scss";
+
+const UserInfoCard = ({
+  user: { id, username, gender, birthday, rating, attitude },
+}) => {
+  const attitudeToUser = getAttitudeToUser(attitude);
+  const linkToUser = `/user/${id}`;
+  const userAge = defineUserAge(birthday);
+  const userGender = genders[gender] || "Not Specified";
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.jsx]: constants

File: src/components/UserInfoCard/UserInfoCard.jsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+  return (
+    <div
+      className="user_info_card"
+      style={{ backgroundColor: attitudeToUser.color }}
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.jsx]: do not use style

File: src/components/UserInfoCard/UserInfoCard.jsx [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+      <div className="user_main_info">
+        <Link to={linkToUser}>{username}</Link>
+        <div className="user_info_item">{userGender}</div>
+        <div className="user_info_item">{userAge}</div>
+      </div>
+
+      <div className="user_rating">
+        <RatingAverage value={rating} direction="col" />
+
+        {[0, 1].includes(attitude) && (
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.jsx]: magic array

File: src/components/UserInfoCard/UserInfoCard.scss [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+.user_info_card {
+  display: flex;
+  flex-direction: column;
+  min-width: 300px;
+  height: 170px;
+  width: $fullWidth;
+  margin: 10px;
+  padding: 10px;
+  border-radius: $r16;
+  box-shadow: 0 0 10px $transparentGreyBackground;
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.scss]: constants

File: src/components/UserInfoCard/UserInfoCard.scss [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    justify-content: center;
+    height: 60px;
+  }
+  .user_main_info {
+    @include flex(column, center, center);
+
+    > :first-child {
+      text-align: center;
+      font-size: $fontsize-24;
+      font-weight: 500;
```
Comment (andrii-kuzmichov) [inline src/components/UserInfoCard/UserInfoCard.scss]: constants

File: src/components/avatar/custom-avatar.js [hunk 17]
```diff
@@ -71,7 +72,7 @@ CustomAvatar.propTypes = {
 CustomAvatar.defaultProps = {
   userId: "",
   name: "",
-  size: "change this",
+  size: "",
   changeAvatarCounter: "change this",
```
Comment (andrii-kuzmichov) [inline src/components/avatar/custom-avatar.js:76]: Drop

File: src/components/helpers/defineUserAge.js [hunk 18]
```diff
@@ -0,0 +1,9 @@
+import getAge from "./get-age-string";
+
+const defineUserAge = birthday => {
+  return Number.isNaN(getAge(birthday)) || !getAge(birthday)
+    ? "Not specified"
```
Comment (andrii-kuzmichov) [inline src/components/helpers/defineUserAge.js]: Reuse from above const

File: src/components/helpers/getAttitudeToUser.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
+const getAttitudeToUser = attitude => {
+  let attitudeToUser = {};
+
+  switch (attitude) {
+    case 0:
+      attitudeToUser = {
+        color: "#c2ffc2",
```
Comment (andrii-kuzmichov) [inline src/components/helpers/getAttitudeToUser.js]: change to className and use for components

File: src/components/helpers/getAttitudeToUser.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+        message: "You like this user",
+        thumb: "fa-thumbs-up",
+      };
+      break;
+    case 1:
+      attitudeToUser = {
+        color: "#ffc6c2",
+        message: "You dislike this user",
```
Comment (andrii-kuzmichov) [inline src/components/helpers/getAttitudeToUser.js]: both message be like as const

File: src/components/users/UserItemList.jsx [hunk 21]
```diff
@@ -1,22 +1,30 @@
...
+import styled from "styled-components";
+import UserInfoCard from "../UserInfoCard/UserInfoCard";
 import PagePagination from "../shared/pagePagination";
 
+const GridLayout = styled.div`
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
+  gap: 10px;
+`;
+
```
Comment (andrii-kuzmichov) [inline src/components/users/UserItemList.jsx]: move to diff file and drop styled components

File: src/components/users/UserItemList.jsx [hunk 22]
```diff
@@ -1,22 +1,30 @@
...
 const UserItemList = ({ page, totalPages, users, callback }) => {
   const handlePageChange = pageEl => {
     callback(
       window.location.search.replace(/(page=)[0-9]+/gm, `page=${pageEl}`),
```
Comment (andrii-kuzmichov) [inline src/components/users/UserItemList.jsx]: Try to refactor it

File: src/components/users/users.scss [hunk 23]
```diff
@@ -0,0 +1,74 @@
+@import "../../styles/main.scss";
+
+.box.card {
+  border-radius: 0.75rem;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
```
Comment (andrii-kuzmichov) [inline src/components/users/users.scss:5]: drop this file

File: src/constants/GenderConstants.js [hunk 24]
```diff
@@ -1,3 +1,3 @@
-﻿const genders = ["Undefined", "Male", "Female", "Other"];
+﻿const genders = ["Not specified", "Male", "Female", "Other"];
```
Comment (andrii-kuzmichov) [inline src/constants/GenderConstants.js]: use constants

File: src/components/UserInfoCard/UserInfoCard.jsx (hunk 9 above)
Comment (ghost) [reply src/components/UserInfoCard/UserInfoCard.jsx:4]: File requires 'export defaut'

**Review by andrii-kuzmichov — DISMISSED** [review]

**Review by andrii-kuzmichov — APPROVED** [review]

**Pull request merged**